src/containers/LoginFormContainer.js

```diff
@@ -1,114 +1,69 @@
 import React, { Component } from "react";
-import FacebookLogin from "react-facebook-login";
-import GoogleLogin from "react-google-login";
-import { BASE_URL, API_FB_ID, API_GOOGLE_ID } from "../constants";
-import { providers, roles } from "../enums.js";
-import axios from "axios";
-import { Redirect } from "react-router-dom";
-import HomePage from "../components/HomePage/HomePage";
-import EditProfileContainer from "./EditProfileContainer";
+import { API_FB_ID, FACEBOOK_SCOPES, API_GOOGLE_ID } from "../constants";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import {
+  loginRequested,
+  handleGoogleSuccess,
+  handleGoogleFailure,
+  handleFacebookResponse
+} from "../actions/LoginActions";
+import LoginForm from "../components/Login/LoginForm";
 
-export default class LoginFormContainer extends Component {
-  state = {
-    isLoggedIn: false,
-    userID: "",
-    firstName: "",
-    lastName: "",
-    email: "",
-    photoUrl: "",
-    externalProviderToken: "",
-    roleName: "",
+class LoginFormContainer extends Component {
+  loginRequested = () => {
+    this.props.actions.loginRequested();
   };
 
-  componentClicked = () => console.log("clicked");
-
-  responseFacebook = response => {
-    console.log(response);
-
-    this.setState({
-      isLoggedIn: true,
-      userID: response.userID,
-      firstName: response.first_name,
-      lastName: response.last_name,
-      email: response.email,
-      photoUrl: response.picture.data.url,
-      externalProviderToken: response.accessToken,
-      externalProviderName: providers.FACEBOOK,
-      roleName: roles.USER
-    });
-
-    this.addNewUser();
-    console.log(response.accessToken);
+  handleFacebookResponse = response => {
+    this.props.actions.handleFacebookResponse(response);
   };
 
-  responseGoogle = response => {
-    console.log(response);
-
-    this.setState({
-      isLoggedIn: true,
-      externalProviderToken: response.accessToken,
-      firstName: response.profileObj.givenName,
-      lastName: response.profileObj.familyName,
-      email: response.profileObj.email,
-      birthday: response.birthday,
-      photoUrl: response.profileObj.imageUrl,
-      externalProviderName: providers.GOOGLE,
-      roleName: roles.USER
-
-    });
-    this.addNewUser();
+  handleGoogleFailure = response => {
```

Comment:
I'm not sure if it's needed, you can use this.props.handleGoogleFailurejust inside render, but you would have to change mapDispatchToProps

Reply:
same as above